src/libcore/iter.rs

```diff
@@ -280,6 +289,16 @@ pub trait Iterator {
     /// different sized integer, the `zip` function provides similar
     /// functionality.
     ///
+    /// # Undefined overflow
+    ///
+    /// The method does no guarding against overflows, so enumerating more than
+    /// `usize::MAX` elements is undefined.
```

Comment:
There was intention, e.g.: https://github.com/rust-lang/rfcs/pull/560#issuecomment-69998193 and https://github.com/rust-lang/rfcs/pull/560#issuecomment-72093071 .

It was even in the text in https://github.com/rust-lang/rfcs/commit/6dbbc70637c3ea550d0458bc7da65348462302f2#diff-3ade9a15caf1d747ea87d08e3021dc33R126 which was edited https://github.com/rust-lang/rfcs/commit/865eae64f376ae4804a52467ebbce31707f3a784#diff-3ade9a15caf1d747ea87d08e3021dc33R129 and then removed in a seemingly unrelated commit https://github.com/rust-lang/rfcs/commit/9f2c7f0e9fe5e4ddffc598bfa3c5f84b9bfcf074 . I'm fairly sure the loss of clarity in the non-`as` case in the last commit was an accident.

> https://internals.rust-lang.org/t/a-tale-of-twos-complement/1062

NB. that was posted a few weeks before the RFC, and the design was refined/changed in the RFC: the RFC is canonical source.

---

That said, I'm in favour of writing this documentation as "behaviour for too-long iterators is unspecified", but not in favour of motivating that by a pedantic reading of the RFC. :)